build_config/lib/src/input_set.dart

```diff
@@ -38,7 +40,8 @@ class InputSet {
           'Expected a Map or a List but got a ${json.runtimeType}');
     }
     final parsed = _$InputSetFromJson(json as Map);
-    if (parsed.include != null && parsed.include!.any((s) => s.isEmpty)) {
+    if (parsed.include != null && parsed.include!.any((s) => s.isEmpty) ||
```

Comment:
You should be able to use `include_defaults` with `excludes` but no `includes`, which I think this won't allow for.

I would also provide a different message for the case where `include_defaults` is set but no other configuration exists, this message is a bit confusing.

Lastly, I would put explicit parenthesis instead of relying on operator precedence here, it will make the intent more clear for the next reader here.

Reply:
@jakemac53 good note about `exclude`. 

I can fix it by two ways. 

**The first**: To set default non null value for `include`, but then `May be null or empty which means every possible path (like '**').` definition will not be true as  `include` will never be null. I found some `include` null checks in the code, so, this change can break something. Also, I don't want to change the established behavior. 

**The second**
I can do something like that:
```
if (parsed.includeDefaults &&
    (parsed.include?.isEmpty ?? true) &&
    (parsed.exclude?.isEmpty ?? true)) {
  throw ArgumentError.value(parsed.includeDefaults, 'include_defaults',
      'Include globs or Exclude globs must be defined if include_defaults is true');
}
if (parsed.includeDefaults && parsed.include == null) {
  // include field could not be null if we would like to include defaults
  parsed =
      InputSet(include: [], exclude: parsed.exclude, includeDefaults: true);
}
```
So, `include` field will never be null if `include_defaults` is true. It allows us to fill `include` with defaults values in the next steps.

I prefer the second way.